goversion/goversion.go

```diff
@@ -156,3 +156,44 @@ func extractPrerelease(part, prerelease *string) {
 		*part = (*part)[:i]
 	}
 }
+
+type GoVersions []*GoVersion
+
+func (versions GoVersions) Len() int      { return len(versions) }
+func (versions GoVersions) Swap(i, j int) { versions[i], versions[j] = versions[j], versions[i] }
+func (versions GoVersions) Less(i, j int) bool {
+	less := func(a, b string) bool {
+		intA, err := strconv.Atoi(a)
+		if err != nil {
+			return false
```

Comment:
This doesn't result in "put non-int-parseable elements at the end of the slice" behavior--this test case fails, for example:

```
		{
			name: "sorting with non-int element",
			versions: GoVersions{
				New("1.2.5"),
				New("1.hi.4"),
				New("1.2.3"),
			},
			expected: GoVersions{
				New("1.2.5"),
				New("1.2.3"),
				New("1.hi.4"),
			},
		},
```

(The result is that the slice order doesn't change at all:
`expected 1.2.3 at index 1, got 1.hi.4`
`expected 1.hi.4 at index 2, got 1.2.3`)

It also isn't transitive:

`1.2.3` .Less `1.hi.2` is false
`1.hi.2` .Less `1.2.1` is false
So you'd expect `1.2.3` .Less `1.2.1` to be false, but it's true.
(I wrote a test case to find these: https://gist.github.com/dagood/9d55f83beb172425fdb45e05949492a2.)

I think this behavior is *ok* with a doc describing it. Leaving it unexported/internal seems even better to me though. I'll push a change to move it.